src/tests/functional/plugin/gpu/single_layer_tests/dynamic/reduce.cpp

```diff
@@ -0,0 +1,265 @@
+            for (const auto& item : shape.second) {
+                result << CommonTestUtils::vec2str(item) << "_";
+            }
+        }
+        result << "axes=";
+        result << CommonTestUtils::vec2str(axes) << "_";
+
+        result << "Precision=" << netType << "_";
+        result << "reductionType=" << reductionType << "_";
+        result << "keepDims=" << keepDims << "_";
+        result << "trgDev=" << targetDevice;
+
+        return result.str();
+    }
+
+protected:
+    void SetUp() override {
+        ReduceInput input_data;
+        ElementType netPrecision;
+        helpers::ReductionType reductionType;
+        bool keepDims;;
+        std::tie(input_data, netPrecision, reductionType, keepDims, targetDevice) = this->GetParam();
+
+        std::vector<InputShape> inputShapes = input_data.data_shape;
+        std::vector<int> axes = input_data.axes;
+
+        init_input_shapes(inputShapes);
+
+        auto params = ngraph::builder::makeDynamicParams(netPrecision, inputDynamicShapes);
+        auto paramOuts = ngraph::helpers::convert2OutputVector(
+                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        std::vector<size_t> shapeAxes;
+        shapeAxes.push_back(axes.size());
+
+        auto reductionAxesNode = std::dynamic_pointer_cast<ngraph::Node>(
+                std::make_shared<ngraph::opset3::Constant>(ngraph::element::Type_t::i64, ngraph::Shape(shapeAxes), axes));
+
+        const auto reduce = ngraph::builder::makeReduce(paramOuts[0], reductionAxesNode, keepDims, reductionType);
+
+        auto makeFunction = [](ParameterVector &params, const std::shared_ptr<Node> &lastNode) {
+            ResultVector results;
+
+            for (int i = 0; i < lastNode->get_output_size(); i++)
+                results.push_back(std::make_shared<opset1::Result>(lastNode->output(i)));
+
+            return std::make_shared<Function>(results, params, "ReduceLayerGPUTest");
+        };
+
+        function = makeFunction(params, reduce);
+    }
+};
+
+TEST_P(ReduceLayerGPUTest, CompareWithRefs) {
+   SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+   run();
+}
+
+namespace {
+
+const std::vector<bool> keepDims = {
+    true,
+    false,
+};
+
+const std::vector<ElementType> floatPrecisions = {
+    ElementType::f32,
+    ElementType::f16,
+};
+
+const std::vector<ElementType> floatIntPrecisions = {
+    ElementType::f32,
+    ElementType::f16,
+    ElementType::i32,
+};
+
+
+
+namespace Reduce {
+
+const ReduceInput dyn1d = {
+    {
+        { {-1}, {{4}, {5}, {3}} }
+    },
+    {0}
+};
+
+const ReduceInput dyn2d = {
+    {
+        { {-1, -1}, {{4, 4}, {5, 5}, {6, 1}} }
+    },
+    {1}
+};
+
+const ReduceInput dyn3d = {
+    {
+        { {-1, -1, -1}, {{4, 4, 4}, {5, 1, 5}, {6, 6, 1}} }
+    },
+    {0, 2}
+};
+
+const ReduceInput dyn4d = {
+    {
+        { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}, {3, 4, 5, 6}} }
+    },
+    {1, 2}
+};
+
+const ReduceInput dyn5d = {
```

Comment:
Reduce operation shows more difference in testing different reducing axis than reduce type.
How about adding tests for variant reduce-axis?
Not to make too many TCs, it would be nice to make TCs of variant axes with one mode.

Reply:
Good idea. I will add various axis with one mode.